src/common/date-provider.ts

```diff
@@ -4,7 +4,11 @@ import { utc } from 'moment';
 
 export class DateProvider {
     public static getDateFromTimestamp(timestamp: string): Date {
-        return new Date(timestamp);
+        const dateNumber = Number(timestamp);
```

Comment:
At first read, I got a little confused with this code and when one and the other scenario happen. I got a better understanding by looking at the unit tests. I think it's like this: on scenario is when we get a string formatted as a date and the other one is wen we get and actual timestamp (like a plain number, usually a lot of digits).
Would it be better if we have some private helper functions with names that help to better understand why are we checking what we are checking?